test/utilsTest.ts

```diff
@@ -0,0 +1,137 @@
+                const file = Utils.findFileByPrioritizedPatterns([], []);
+                should(file).be.null;
+            });
+        });
+
+        context('when subStrings is null', () => {
+            it('should return null', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['some-file.txt'], null);
+                should(file).be.null;
+            });
+        });
+
+        context('when subStrings is empty array', () => {
+            it('should return null', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['some-file.txt'], []);
+                should(file).be.null;
+            });
+        });
+
+        context('when subStrings matches the first file', () => {
+            it('should return the first file', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['a', 'b', 'c'], [['a']]);
+                should(file).be.equal('a');
+            });
+        });
+
+        context('when subStrings matches last file', () => {
+            it('should return the last file', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['a', 'b', 'c'], [['c']]);
+                should(file).be.equal('c');
+            });
+        });
+
+        context('when subStrings matches a file', () => {
+            it('should return the matched file', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['a', 'bc', 'de'], [['c']]);
+                should(file).be.equal('bc');
+            });
+        });
+
+        context('when subStrings includes more then one sub-string', () => {
+            it('should return the file that matches all the substrings', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['b', 'c', 'bc'], [['c', 'b']]);
+                should(file).be.equal('bc');
+            });
+        });
+
+        context('when subStrings includes more then one sub-string', () => {
+            it('should return the file that matches all the substrings', () => {
+                const file = Utils.findFileByPrioritizedPatterns(['b', 'c', 'bc'], [['c', 'b']]);
+                should(file).be.equal('bc');
+            });
+        });
+
+        context('when only the last sub-strings matches a file', () => {
+            it('should return the file that matches the last substrings', () => {
+                const file = Utils.findFileByPrioritizedPatterns(
+                    ['cbd', 'c'], 
+                    [['c', 'b'], ['e'], ['c','b','d']]);
+                should(file).be.equal('cbd');
+            });
+        });
+
+        context('when files include paths', () => {
+            it('should only compare based on file name and ignore the rest of the path', () => {
+                const file = Utils.findFileByPrioritizedPatterns(
+                    ['d/ba', 'd/eb', 'a/bd'], [['b', 'd']]);
+                should(file).be.equal('a/bd');
+            });
+        });
```

Comment:
should add a test to check that the first matching file is returned, if there are multiple that match the same patterns

Reply:
Currently it is in a don't care state. I don't think returning the first one is better then returning the last one.